librz/core/cmd/cmd_print.c

```diff
@@ -6599,3 +6511,72 @@ RZ_IPI RzCmdStatus rz_print_byte_bitstream_handler(RzCore *core, int argc, const
 	free(str_buf);
 	return RZ_CMD_STATUS_OK;
 }
+
+RZ_IPI RzCmdStatus rz_cmd_print_asn1_handler(RzCore *core, int argc, const char **argv) {
+	RASN1Object *asn1 = rz_asn1_create_object(core->block, core->blocksize);
+	if (!asn1) {
+		RZ_LOG_ERROR("core: Malformed object: did you supply enough data?\ntry to change the block size (see b?)\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	char *res = rz_asn1_to_string(asn1, 0, NULL);
+	rz_asn1_free_object(asn1);
+	if (!res) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_printf("%s\n", res);
+	free(res);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_print_protobuf_handler(RzCore *core, int argc, const char **argv) {
+	char *s = rz_protobuf_decode(core->block, core->blocksize, true);
+	if (!s) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_printf("%s", s);
+	free(s);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_print_pkcs7_handler(RzCore *core, int argc, const char **argv) {
+	RCMS *cms = rz_pkcs7_parse_cms(core->block, core->blocksize);
+	if (!cms) {
+		RZ_LOG_ERROR("core: Malformed object: did you supply enough data?\ntry to change the block size (see b?)\n");
+	}
+	char *res = rz_pkcs7_cms_to_string(cms);
+	if (res) {
+		rz_cons_printf("%s\n", res);
+	}
+	free(res);
+	rz_pkcs7_free_cms(cms);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_print_x509_handler(RzCore *core, int argc, const char **argv) {
+	RASN1Object *object = rz_asn1_create_object(core->block, core->blocksize);
+	RX509Certificate *x509 = rz_x509_parse_certificate(object);
+	if (!x509) {
+		RZ_LOG_ERROR("core: Malformed object: did you supply enough data?\ntry to change the block size (see b?)\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzStrBuf *sb = rz_strbuf_new("");
+	rz_x509_certificate_dump(x509, NULL, sb);
+	char *res = rz_strbuf_drain(sb);
+	if (res) {
+		rz_cons_printf("%s\n", res);
+		free(res);
+	}
+	rz_x509_free_certificate(x509);
```

Comment:
weird naming convention here :(

Reply:
I will fix this today

Reply:
My MR should be merged before this one: https://github.com/rizinorg/rizin/pull/3077